packages/core/src/routes/hook.ts

```diff
@@ -23,11 +26,28 @@ export default function hookRoutes<T extends AuthedRouter>(
 
   router.post(
     '/hooks',
-    koaGuard({ body: Hooks.createGuard.omit({ id: true }), response: Hooks.guard, status: 200 }),
+    koaGuard({
+      body: Hooks.createGuard
+        .omit({ id: true, event: true, enabled: true })
+        .extend({ config: hookConfigGuard.omit({ signingKey: true }) }),
+      response: Hooks.guard,
+      status: [200, 422],
+    }),
     async (ctx, next) => {
+      const { config, events, ...rest } = ctx.guard.body;
+      assertThat(events.length > 0, new RequestError({ code: 'hook.empty_events', status: 422 }));
+
+      // Note: the `event` is deprecated, but we still need to set it for backward compatibility.
+      const event = events[0];
+      assertThat(event !== undefined, new Error('Cannot set up event for backward compatibility'));
```

Comment:
if the type of the events is guarded by the hookEventGuard and events.length > 0 is guarded above, events[0] will never be empty?

Reply:
Cool, updated.